src/views/Products/ProductListing.jsx

```diff
@@ -0,0 +1,38 @@
+import useMainState from "../../contexts/MainContextProvider";
+import SingleProduct from "../../components/ProductCard";
+// todo later : when importing directly, it's showing error 
+import { useSearchParams } from "../../../node_modules/react-router-dom/index";
+import React from "react";
+
+
+const style = {
+  display: "flex",
+  flexWrap: "wrap",
+  maxWidth: "900px",
+  justifyContent: "center",
+  marginLeft: "auto",
+  marginRight: "auto",
+};
+
+const ProductListing = () => {
+  const {
+    state: { products },
+  } = useMainState();
+  console.log(products);
+
+  let [searchParams] = useSearchParams(); 
+  const searchTerm = searchParams.get('search')?.toLowerCase();
+  
+  const filteredProducts = searchTerm ? products.filter((product) => product.name.toLowerCase().includes(searchTerm) || product.description.toLowerCase().includes(searchTerm)) : products;
```

Comment:
make it a function or use if else statements to make it readable, we should use ternary operator only for single line statements